### ISSUE TITLE
Problem when running spec suites from CLI

### ISSUE THREAD

**Issue by mirfilip** [phpspec/phpspec#641]
After updating from `2.0.*` to `2.1.*`, there is a problem when running specs from CLI.

# Project details
Directory structure
```
+-- phpspec.yml.dist
+-- module
|   +-- Api
|        +-- src
|   +-- Ovp
|        +-- src
+-- tests
|   +-- unit
|   +-- features
|   +-- spec
|        +-- Api
|        +-- Ovp
```
composer.json autoloading part
```
"autoload": {
        "psr-0": {
            "Api": "module/Api/src",
            "Ovp": "module/Ovp/src"
        }
    },
    "autoload-dev": {
        "psr-4": {
            "Behat\\": "tests/features/src"
        },
        "psr-0": {
            "": "tests/unit"
        }
    }
```

phpspec.yml.dist
```
code_generator.templates.paths:
  - tests/spec/templates

suites:
  ovp:
    namespace: Ovp
    src_path: module/Ovp/src
    spec_path: tests
  api:
    namespace: Api
    src_path: module/Api/src
    spec_path: tests

formatter.name: pretty
```

# Problem
Running single test from CLI using `2.0.*`:
```
> pwd // I'm in a project root
> phpspec run tests/spec/Api/Config/ProviderContextSpec.php // One of existing tests
> (...)
1 specs
2 examples (1 passed, 1 broken)
14ms
```

Running single test from CLI using `2.1.*`:
```
> pwd // I'm in a project root
> phpspec run tests/spec/Api/Config/ProviderContextSpec.php // One of existing tests
> (...)
[RuntimeException]
  Spec class must be in the base spec namespace `spec\Ovp\`.
```
I noticed that when I comment out `ovp` suite, everything works well.
What happens ? Is there anything wrong with suites configuration ?

**Comment by mirfilip** [comment]
Funnily enough, I've also noticed that running `phpspec` against **source file** not **spec file** works well
```
> phpspec run module/Api/src/Api/Config/ProviderContext.php
> (...)
1 specs
5 examples (5 passed)
16ms
```

**Comment by ciaranmcnulty** [comment]
What namespace is the spec you're trying to run in?

**Comment by mirfilip** [comment]
```
<?php
namespace spec\Api\Config;
(...)
```
@ciaranmcnulty

**Comment by mirfilip** [comment]
Any idea what can go wrong here ?

**Comment by ciaranmcnulty** [comment]
It's possible that the duplicate spec_path is the issue. PhpSpec is trying to figure out from the spec filename which suite it's in and having issues.

**Comment by mirfilip** [comment]
Is it a configuration problem or a bug ? If former, how can I accomplish having multiple suites ?

**Comment by fojuth** [comment]
The bug still exists, making it impossible to run multiple suite tests. Furthermore, if I comment-out the suite the problem is connected to, phpspec can't run tests that are combined with the commented-out suite (which is quite obvious, still, maybe it should ignore these tests?). This behaviour renders the whole process useless since there is no command (at least none that I know of) making it possible to run tests per suite.

**Comment by fojuth** [comment]
The problem is probably with the `PSR0Locator` class (or the IoC).

The `createResourceFromSpecFile` method checks if the spec is in the correct namespace. The spec's path is correct but the whole `PSR0Locator` is not. It "exists" in the previous suite context although the `ContainerAssembler` creates a proper instance of the `PSR0Locator`.

This means that the `PSR0Locator` checks the spec's namespace against it's own improper context. Forgive me, if my explanation is a bit vague ;)

When I commented-out the fragment below (`PSR0Locator::createResourceFromSpecFile()`) everything started to work fine:

```php
if (0 !== strpos($classname, $specNamespace)) {
    throw new \RuntimeException(sprintf(
        'Spec class `%s` must be in the base spec namespace `%s`.',
        $classname,
        $this->getSpecNamespace()
    ));
}
```

The `$classname` is ok, but `$this->getSpecNamespace()` returns wrong namespace (previous test suite). Probably the `PSR0Resource` is created correctly, that's why it works properly.

This is not a fix, it's just a temporary solution. Still, it works and allows creating multiple test suites. Hope this helps you guys :)

**Comment by mirfilip** [comment]
I'm not sure if it was fully fixed but the problem doesn't occur in **2.4.1**. @ciaranmcnulty you may want to revisit & close this.

**Comment by ciaranmcnulty** [comment]
Oh weird, thanks!
